src/test/java/ie/gmit/softwareeng/assignment/Tests.java

```diff
@@ -0,0 +1,98 @@
+package ie.gmit.softwareeng.assignment;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class Tests {
+
+    FilmService filmService;
+    int id;
+
+    @Mock
+    FilmRepository repository;
+
+    @Mock
+    OmdbClient omdbClient;
+
+    @Mock
+    Film film;
+
+    @Before
+    public void setup(){
+        filmService = new FilmService(repository, omdbClient);
+        id = 1;
+        film = new Film(id, "Eds film :)", 5);
+    }
+
+    @Test(expected = FilmNotFoundException.class)
+    public void testGetFilmInvalidFilm(){
+        when(repository.getFilm(id)).thenReturn(null);
+        filmService.getFilm(id);
+    }
+
+    @Test(expected = FilmNotFoundException.class)
+    public void testRentFilmInvalidFilm(){
+        when(repository.getFilm(id)).thenReturn(null);
+        filmService.rentFilm(id);
+    }
+
+    @Test(expected = NotInStockException.class)
+    public void testRentFilmNoStock(){
+        Film film2 = new Film(2, "Eds film 2 electric boogaloo", 0);
+        when(repository.getFilm(id)).thenReturn(film2);
+//        when(film.getCopiesInStock()).thenReturn(0); // why doesn't this work ? :(
```

Comment:
You are instantiating a new film in setup, which removes the mock that was created by Mockito. If you remove that, it should use the mockito mock.